File: sharedlife-frontend/src/app/services/spent/spent.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class SpentService {

  private baseURL = "http://localhost:8080/spents";

  constructor(
    private HttpClient: HttpClient
  ) { }

  createSpent(spent: any, admin: boolean){
    return this.HttpClient.post(`${this.baseURL}/create?admin=${admin}`, spent);
  }

  getAllDebtsByUsername(username: string){
    return this.HttpClient.get<any>(`${this.baseURL}/debts?username=${username}`);
  }

  getSpentById(id: string){
    return this.HttpClient.get<any>(`${this.baseURL}/id/${id}`);
  }

  getSpentsByUsernameAndHomeId(username: string, id: string){
    return this.HttpClient.get<any>(`${this.baseURL}/home/${id}/username?username=${username}`);
  }

  getDebtsBySpentId(id: string){
    return this.HttpClient.get<any>(`${this.baseURL}/debts/id/${id}`);
  }

  getAllSpentByHomeId(id: string){
    return this.HttpClient.get<any>(`${this.baseURL}/byHomeId/${id}`);
  }
  // @PutMapping("/debt/{id}/paid")
  paidDebt(id: string, paid: boolean){
    return this.HttpClient.put(`${this.baseURL}/debt/${id}/paid`, paid);
  }

  deleteSpentAndDebts(id: string){
    return this.HttpClient.delete(`${this.baseURL}/${id}/delete`);
  }

  verifyUserPaidDebt(idDebt: string, username:string){
    return this.HttpClient.get<any>(`${this.baseURL}/debt/${idDebt}/username?username=${username}`)
  }

  getDebtsInfo(idSpent: string){
    return this.HttpClient.get<any>(`${this.baseURL}/${idSpent}/debtsInfo`);
  }

}

File: sharedlife-backend/src/main/java/net/tfg/sharedlife/service/spent/SpentService.java
package net.tfg.sharedlife.service.spent;

import java.util.List;

import net.tfg.sharedlife.dto.DebtDTO;
import net.tfg.sharedlife.dto.SpentCheckPaidDTO;
import net.tfg.sharedlife.dto.SpentDTO;

public interface SpentService {
	
	void createSpent(SpentDTO spentDto, boolean admin);
	
	List<DebtDTO> getDebtsByUsername(String username);
	
	SpentDTO getSpentById(Long id);
	
	List<SpentDTO> getSpentsByUsernameAndHomeId(String username, Long id);
	
	List<DebtDTO> getDebtsBySpentId(Long id);
	
	List<SpentDTO> getAllSpentsByHomeId(Long id);
	
	void updatePaidStatus(Long id, boolean paid);
	
	void deleteSpentAndDebts(Long id);
	
	void deleteDebt(Long id);

	void deleteSpent(Long id);

	boolean verifyUserPaidDebt(Long idDebt, String username);

	SpentCheckPaidDTO getDebtsInfo(Long idSpent);
	
}
